adr/0011-discovery-requires-unique-id.md

```diff
@@ -0,0 +1,45 @@
+# 0011. Discovery Requires Unique ID
+
+Date: 2020-04-27
+
+## Status
+
+Accepted
+
+## Context
+
+There is a mechanism to ignore discovered config flows using unique ids attached to config flow
+but not all integrations may set an unique id due to lack of truly unique identifiers
+or just not having been updated for concerns of backwards compatibility.
+
+This creates an UX issue that has started to become more pronounced with more config flows
+being added every release (which is awesome!). The real issue is not all discovery processes 
+are created equal and that becomes an issue for consistency.
+
+## Proposal
+
+All new integratons with discoverable config flows should be required to set an
+unique id during discovery.
+
+All existing integrations with discoverable config flows should be reviewed for
+possible ways to improve identification during discovery. A whitelist could be developed
+to allow existing integrations to remain functional under new requirements.
+
+If an unique id can sometimes not be available such as older devices that lack unique identifiers
+in zeroconf/ssdp/api/etc then the config entry should not be shown to user.
```

Comment:
```suggestion
If a unique id can not be made available, such as on older devices that lack unique identifiers
in zeroconf/ssdp/api/etc, then the config entry should not be shown to user.
```

Reply:
Edited your suggestion, as SSDP, API should be capitalized as well and `the` was missing before `user`.